test/components/tableTests.ts

```diff
@@ -1,288 +1,112 @@
-      let t = new Plottable.Components.Table([[c1]]);
-
-      assert.throw(() => t.add(null, 0, 0), "Cannot add null to a table cell");
+  function assertTableRows(table: Plottable.Components.Table, expectedRows: Plottable.Component[][], message: string) {
+    expectedRows.forEach((row, rowIndex) => {
+      row.forEach((component, columnIndex) => {
+        assert.strictEqual(table.componentAt(rowIndex, columnIndex), component,
+          message + ` contains the correct entry at [${rowIndex}, ${columnIndex}]`);
+      });
     });
+  }
 
-    it("add()-ing a Component to the Table should detach() it from its current location", () => {
-      let c1 = new Plottable.Component;
-      let svg = TestMethods.generateSVG();
-      c1.renderTo(svg);
+  describe("constructor behavior", () => {
+    it("adds the correct CSS class", () => {
       let table = new Plottable.Components.Table();
-      table.add(c1, 0, 0);
-      assert.isFalse((<Node> svg.node()).hasChildNodes(), "Component was detach()-ed");
-      svg.remove();
+      assert.isTrue(table.hasClass("table"));
     });
 
-    it("add() works even if a component is added with a high column and low row index", () => {
-      // Solves #180, a weird bug
-      let t = new Plottable.Components.Table();
-      let svg = TestMethods.generateSVG();
-      t.add(new Plottable.Component(), 1, 0);
-      t.add(new Plottable.Component(), 0, 2);
-      t.renderTo(svg); // would throw an error without the fix (tested);
-      svg.remove();
+    it("can take a list of lists of Components in the constructor", () => {
+      let c00 = new Plottable.Component();
+      let c11 = new Plottable.Component();
+      let rows = [
+        [c00, null],
+        [null, c11]
+      ];
+      let table = new Plottable.Components.Table(rows);
+
+      assertTableRows(table, rows, "constructor initialized Table correctly");
     });
   });
 
-  it("basic table with 2 rows 2 cols lays out properly", () => {
-    let tableAndcomponents = generateBasicTable(2, 2);
-    let table = tableAndcomponents.table;
-    let components = tableAndcomponents.components;
-
-    let svg = TestMethods.generateSVG();
-    table.renderTo(svg);
-
-    let elements = components.map((r) => (<any> r)._element);
-    let translates = elements.map((e) => TestMethods.getTranslate(e));
-    assert.deepEqual(translates[0], [0, 0], "first element is centered at origin");
-    assert.deepEqual(translates[1], [200, 0], "second element is located properly");
-    assert.deepEqual(translates[2], [0, 200], "third element is located properly");
-    assert.deepEqual(translates[3], [200, 200], "fourth element is located properly");
-    let bboxes = elements.map((e) => Plottable.Utils.DOM.elementBBox(e));
-    bboxes.forEach((b) => {
-      assert.strictEqual(b.width, 200, "bbox is 200 pixels wide");
-      assert.strictEqual(b.height, 200, "bbox is 200 pixels tall");
+  describe("checking Table contents", () => {
+    describe("has()", () => {
```

Comment:
Is there any value to having a nested describe block here rather than either just an `it` or just a single `describe`?